torch/_dynamo/backends/distributed.py

```diff
@@ -146,6 +146,25 @@ def has_higher_order_op(gm):
     return False
 
 
+def propagate_metadata(orig_gm, split_gm) -> None:
+    for _, module in split_gm.named_modules():
+        # TODO: add split id to CompileId: https://github.com/pytorch/tlparse/pull/83/files#r1880649384
```

Comment:
Need to add split id to CompileId to make sure each graph has a unique id. This should be similar to [compiled autograd id](https://github.com/pytorch/pytorch/pull/141907)